File: Assets/MAPNAV/Scripts/GPS-Status.js
#pragma strict
@script AddComponentMenu ("MAPNAV/GPS-Status")
//MAPNAV Navigation ToolKit v.1.0
//This script is for illustrative purposes only. Feel free to modify, extend or customize it to fit your own needs.

var refreshRate : float=0.2;
var style : GUIStyle;
var style2 : GUIStyle;
private var gps : MapNav;
private var ddLat : String;
private var ddLon : String;
private var dmsLat : String;
private var dmsLon : String;
private var heading : float;
private var error : float;
private var status : String;
private var screenX : int;
private var screenY : int;
private var zoom : int;
private var altitude :float;
private var info : boolean;

function Awake(){
	//Reference to MapNav.js script. Make sure that the map object containing the MapNav.js script is tagged as "GameController"
	gps = GameObject.FindGameObjectWithTag("GameController").GetComponent(MapNav);
	screenX=Screen.width;
	screenY=Screen.height;
}

//Get gps Status Data every "refreshRate" seconds
InvokeRepeating("GetData",1.0,refreshRate);

function GetData(){
	//Current latitude (decimal)
	ddLat = gps.userLat.ToString();
	//Current longitude (decimal)
	ddLon = gps.userLon.ToString();
	//Current latitude (degrees, minutes, seconds)
	dmsLat = gps.dmsLat;
	//Current longitude (degrees, minutes, seconds)
	dmsLon = gps.dmsLon;
	//Current heading/orientation
	heading=gps.heading;
	//Current GPS sensor accuracy
	error=gps.accuracy;
	//Current Zoom Level
	zoom=gps.zoom;
	//Current altitude(meters)
	altitude=gps.altitude;
}

function Update(){
	//Reference to MapNav.js "status" variable  
	status=gps.status;
	//Reference to MapNav.js "info" variable. Used to activate/de-activate the GUI elements.
	info=gps.info;
}

function OnGUI () {
	if(info){
		//These GUI Styles can be modified using the inspector
		style.fontSize=(screenX+screenY)*0.015;
		style2.fontSize=(screenX+screenY)*0.015;
		//Display current gps Status data
		GUI.BeginGroup (Rect (0,0, screenX, screenY/4));
		GUI.Box (Rect (0,0,screenX, screenY/4), "");
		GUI.Label(Rect(screenX/40,screenY/50,screenX-screenX/20,screenY/50),"Latitude: "+ dmsLat,style);
		GUI.Label(Rect(screenX/40,3*screenY/50,screenX-screenX/20,screenY/50),"Longitude: "+ dmsLon,style);
		GUI.Label(Rect(screenX/40,5*screenY/50,screenX-screenX/20,screenY/50),"Altitude(m): "+ altitude,style);
		
		GUI.Label(Rect(screenX/40,screenY/50,screenX-screenX/20,screenY/50),"Heading: "+ Mathf.Round(heading),style2);
		GUI.Label(Rect(screenX/40,3*screenY/50,screenX-screenX/20,screenY/50),"Zoom Level: "+ zoom,style2);
		GUI.Label(Rect(screenX/40,5*screenY/50,screenX-screenX/20,screenY/50),"Error(m): "+ error,style2);
		
		GUI.Label(Rect(screenX/40,7*screenY/50,screenX-screenX/20,screenY/25), "Status: "+status,style);
		GUI.EndGroup ();
	}
}

File: Assets/MAPNAV/Scripts/GPS_Status.cs
//MAPNAV Navigation ToolKit v.1.3.2
//This script is for illustrative purposes only. Feel free to modify, extend or customize it to fit your own needs.

using UnityEngine;
using System.Collections;
[AddComponentMenu("MAPNAV/GPS_Status")]

public class GPS_Status : MonoBehaviour
{
    public float refreshRate = 0.2f;
    public GUIStyle style;
	public GUIStyle style2;
    private MapNav gps;
    private string ddLat;
    private string ddLon;
    private string dmsLat;
    private string dmsLon;
    private float heading;
    private float error;
    private string status;
    private int screenX;
    private int screenY;
    private int zoom;
    private float altitude;
    private bool info;

    void Awake()
    {
        //Reference to MapNav.cs script. Make sure that the map object containing the MapNav.cs script is tagged as "GameController"
        gps = GameObject.FindGameObjectWithTag("GameController").GetComponent<MapNav>();
        screenX = Screen.width;
        screenY = Screen.height;
    }

    void Start()
    {
        //Get gps Status Data every "refreshRate" seconds
        InvokeRepeating("GetData", 1.0f, refreshRate);
    }

    void GetData()
    {
        //Current latitude (decimal)
        ddLat = gps.userLat.ToString();
        //Current longitude (decimal)
        ddLon = gps.userLon.ToString();
        //Current latitude (degrees, minutes, seconds)
        dmsLat = gps.dmsLat;
        //Current longitude (degrees, minutes, seconds)
        dmsLon = gps.dmsLon;
        //Current heading/orientation
        heading = gps.heading;
        //Current GPS sensor accuracy
        error = gps.accuracy;
        //Current Zoom Level
        zoom = gps.zoom;
        //Current altitude(meters)
        altitude = gps.altitude;
    }

    void Update()
    {
        //Reference to MapNav.cs "status" variable  
        status = gps.status;
        //Reference to MapNav.cs "info" variable. Used to activate/de-activate the GUI elements.
        info = gps.info;
    }

    void OnGUI()
    {
        if (info)
        {
            //These GUI Styles can be modified using the inspector
            style.fontSize = (int) Mathf.Round((screenX + screenY) * 0.015f);
            style2.fontSize = (int) Mathf.Round((screenX + screenY) * 0.015f);

            //Display current gps Status data
            GUI.BeginGroup(new Rect(0, 0, screenX, screenY/4));
			GUI.Box(new Rect(0, 0, screenX, screenY/4), "");
			GUI.Label(new Rect(screenX/40, screenY/50, screenX-screenX/20, screenY/50), "Latitude: " + dmsLat, style);
			GUI.Label(new Rect(screenX/40, 3*screenY/50, screenX-screenX/20, screenY/50), "Longitude: " + dmsLon, style);
			GUI.Label(new Rect(screenX/40, 5*screenY/50, screenX-screenX/20, screenY/50), "Altitude(m): " + altitude, style);
			GUI.Label(new Rect(screenX/40, screenY/50, screenX-screenX/20, screenY/50), "Heading: " + Mathf.Round(heading), style2);
			GUI.Label(new Rect(screenX/40, 3*screenY/50, screenX-screenX/20, screenY/50), "Zoom Level: " + zoom, style2);
			GUI.Label(new Rect(screenX/40, 5*screenY/50, screenX-screenX/20, screenY/50), "Error(m): " + error, style2);
			GUI.Label(new Rect(screenX/40, 7*screenY/50, screenX-screenX/20, screenY/25), "Status: " + status, style);
            GUI.EndGroup();
        }
    }
}